### DISCUSSION TITLE
Support for non Magisk super user in ReVanced Manager

### DISCUSSION THREAD

**Discussion by ghost** [ReVanced/revanced-discussions#172] [General]
i used to have an issue with Vanced where i can't install root-variant with other alternative superuser managers such as phh superuser or superSU

vanced root variant was magis specific, which was not good cuz of their not-so-good libsu implementation

can this not happen with revanced?

**Comment by Degl1** [comment]
I think that ReVanced will stay to FOSS and non-outdated for their root implementation. But anyone in the team correct me.

**Comment by oSumAtrIX** [comment]
We are using https://github.com/topjohnwu/libsu, the same as Vanced. If you can find an appropriate library for your case, we can try to add it along side.

**Comment by ghost** [comment]
> We are using https://github.com/topjohnwu/libsu, the same as Vanced. If you can find an appropriate library for your case, we can try to add it along side. @oSumAtrIX 

can you try supporting this? https://github.com/seSuperuser/Superuser

**Comment by oSumAtrIX** [comment]
Ideally we need a java library/wrapper, unless we plan to call native functions

**Comment by ghost** [comment]
@oSumAtrIX i have no idea what are you saying. but sure

**Comment by Degl1** [comment]
I would only question the security and integrity of the project. As the previous commit was nearly 5 years ago.

**Comment by Sculas** [comment]
I personally agree with ps1-hagrid. Don't you think it's time to upgrade? Superuser might be outdated, full of bugs, and/or contain vulnerabilities.

**Comment by Degl1** [comment]
Yeah no reason to use alternative root solutions unless you're on Android 4.4.4. Magisk is the way in the current world. And if you're on 4.4.4 you got bigger problems too(no Google login).

**Comment by oSumAtrIX** [comment]
> I personally agree with ps1-hagrid. Don't you think it's time to upgrade? Superuser might be outdated, full of bugs, and/or contain vulnerabilities.

I have consulted him, he is forced to use these alternatives due to his phone

**Comment by Degl1** [comment]
Well there seems to be no solution as all the alternate SU libraries are very out of date.

**Comment by oSumAtrIX** [comment]
There is a solution, but it's not elegant. Technically you can execute all the commands via a system call. Basically doing `system("su -c CMD")`. If anyone wants, they can implement an alternate way like that along the current way.

**Comment by Degl1** [comment]
Sounds quite janky. But if there is request for it. It should be there.

**Comment by lotusdroid** [comment]
System partition in Android 10+ is read-only.
Only Magisk method will work on Android 10+ unless you mount it as read-write.
Less devices are running Android 9 or lower but it would be nice if It support other method.
🥰

**Comment by Degl1** [comment]
In the worst case you can just use OverlayFS.

**Reply by tillnelown** [reply]
No, there are a lot of devices that don't have OverlayFS in the kernel, this is also why Magisk uses bind mounts instead of OverlayFS because mount is so basic no one removes it/it'd take extra work to remove it.
